colearn_keras/test_keras_learner.py

```diff
@@ -28,6 +28,8 @@ def get_mock_model() -> mock.Mock:
     model.evaluate.return_value = {"loss": 1,
                                    "accuracy": 3}
     model.get_weights.return_value = "all the weights"
+    model.optimizer.get_config.return_value = {"name": "Adam"}
```

Comment:
why we need this here? do this PR changes the overall 'how to set up learner' pipeline?

Reply:
No, that is not changed (the demos are still running without any modification). 

The `model` had an `optimizer` attribute which had a `get_config()` function before too. 
But is was not used, and it is used now after these commits. However, the `Mock()` object in the tests gave a `'Mock' object is not subsciptable` error when I wanted to access `model.optimizer.get_config()['name']`, where `model` is a `Mock` object. And to run the function, which resets the optimizer I need to get that attribute.

That is the reason for this line of code.

Reply:
okay